api/hypershift/v1alpha1/hostedcluster_types.go

```diff
@@ -1574,6 +1574,46 @@ type AzurePlatformSpec struct {
 	SubnetID          string `json:"subnetID"`
 	SubscriptionID    string `json:"subscriptionID"`
 	SecurityGroupID   string `json:"securityGroupID"`
+
+	// MSIClientIDs contains the client IDs related to the managed identities needed for the following control plane
+	// components: cluster-image-registry, cluster-ingress, cluster-storage, and cluster-network operators.
+	//
+	// +optional
+	MSIClientIDs *ControlPlaneManagedServiceIdentities `json:"msiClientIDs,omitempty"`
+}
+
+type ControlPlaneManagedServiceIdentities struct {
+	// ImageRegistryMSIClientID is the client ID of a pre-existing managed identity ID of that will be associated with
+	// the cluster-image-registry-operator. The managed identity will be in a different resource group other than
+	// ResourceGroupName.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ImageRegistryMSIClientID string `json:"imageRegistryMSIClientID,omitempty"`
```

Comment:
required fields should not have `omitempty`
same comment for the rest of the fields.